src/main/java/net/mcreator/generator/template/TemplateExpressionParser.java

```diff
@@ -123,17 +97,41 @@ private static boolean parseCondition(@Nonnull Generator generator, @Nonnull Str
 		return false;
 	}
 
-	private static boolean processFTLExpression(Generator generator, String expression) {
+	private static Object getValueFrom(String field, Object conditionDataProvider) throws ReflectiveOperationException {
+		if (!field.contains("()")) { // field
+			return conditionDataProvider.getClass().getField(field.trim()).get(conditionDataProvider);
+		} else { // method
+			return conditionDataProvider.getClass().getMethod(field.replace("()", "").trim())
+					.invoke(conditionDataProvider);
+		}
+	}
+
+	public static Object processFTLExpression(Generator generator, String expression, Object dataHolder) {
 		try {
-			Template t = new Template("INLINE EXPRESSION", new StringReader(expression),
+			Map<String, Object> dataModel = new HashMap<>(generator.getBaseDataModelProvider().provide());
+			AtomicReference<?> retVal = new AtomicReference<>(null);
+			dataModel.put("retVal", retVal);
+			if (dataHolder != null)
+				dataModel.put("data", dataHolder);
+
+			String expr = !checkStartsWithAnyKey(expression, dataModel) && dataHolder != null ?
+					"data." + expression : // by default, dataHolder is used to process the expression

```

Comment:
If expression is eg. `0 != value`, this will translate to `data.0 != value`, not sure how this is meant to work and how this is robust to any expression?

Reply:
Oh yeah, I forgot this is not the same as non-FTL conditions, so this is probably not even needed.

Reply:
:heavy_check_mark: Fixed